Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/LogExceptionAnalyzer.cs

```diff
@@ -0,0 +1,106 @@
+{
+	/// <summary>
+	/// Report when a catch exception block does not call one of the logging methods.
+	/// </summary>
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class LogExceptionAnalyzer : DiagnosticAnalyzer
+	{
+		private const string Title = "Log caught exceptions.";
+		private const string Message = "Exception that is caught, is not logged.";
+		private const string Description = "Log caught exceptions.";
+		private const string Category = Categories.Maintainability;
+
+		private static readonly DiagnosticDescriptor Rule =
+			new DiagnosticDescriptor(
+				Helper.ToDiagnosticId(DiagnosticIds.LogException),
+				Title,
+				Message,
+				Category,
+				DiagnosticSeverity.Error,
+				isEnabledByDefault: false,
+				description: Description
+			);
+
+		private string logMethodNames;
+
+		/// <summary>
+		/// <inheritdoc/>
+		/// </summary>
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		/// <summary>
+		/// <inheritdoc/>
+		/// </summary>
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction(
+				startContext =>
+				{
+					var additionalFiles = new AdditionalFilesHelper(
+						startContext.Options,
+						startContext.Compilation);
+					var methodNames = additionalFiles.GetValueFromEditorConfig(Rule.Id, "log_method_names");
+					if (string.IsNullOrEmpty(methodNames))
+					{
+						startContext.RegisterCompilationEndAction(ReportParsingError);
+					}
+					else
+					{
+						logMethodNames = methodNames;
+						context.RegisterSyntaxNodeAction(AnalyzeCatchException, SyntaxKind.CatchClause);
+					}
+				});
+		}
+
+		private void AnalyzeCatchException(SyntaxNodeAnalysisContext context)
+		{
+			var catchNode = context.Node;
+			// Look for logging method calls underneath this node.
+			var hasCallingLogNodes = catchNode.DescendantNodes()
+				.OfType<InvocationExpressionSyntax>().Any(x => IsCallingLogMethod(context, x));
+			// If another exception is thrown, logging is not required.
+			var hasThrowNodes = catchNode.DescendantNodes().OfType<ThrowStatementSyntax>().Any();
+			if (!hasCallingLogNodes && !hasThrowNodes)
+			{
+				var location = catchNode.GetLocation();
+				context.ReportDiagnostic(Diagnostic.Create(Rule, location));
+			}
+		}
+
+		private void ReportParsingError(CompilationAnalysisContext context)
+		{
+			var loc = Location.Create(context.Compilation.SyntaxTrees.First(), TextSpan.FromBounds(0, 0));
+			context.ReportDiagnostic(Diagnostic.Create(Rule, loc));
+		}
+
+		private bool IsCallingLogMethod(SyntaxNodeAnalysisContext context, SyntaxNode node)
+		{
+			var additionalFilesHelper = new AdditionalFilesHelper(context.Options, context.Compilation);
+			var logMethodNames = additionalFilesHelper.GetValueFromEditorConfig(Rule.Id, @"log_method_names");
```

Comment:
Remove this?  Maybe rename the field _logMethodNames so it's clear it's a field?

Reply:
Oops, removed it now and renamed the field as you suggusted